core/block.js

```diff
@@ -224,11 +231,35 @@ Blockly.Block.prototype.data = null;
 Blockly.Block.prototype.colour_ = '#000000';
```

Comment:
So I think that it has to be in block because this is where we get all the information from the json object. I think we lose the colour value if we do not store it on the block. However, not a 100% sure so will investigate further.